src/main/java/org/shanerx/tradeshop/objects/ShopItemStack.java

```diff
@@ -227,6 +232,8 @@ public boolean isSimilar(ItemStack toCompare) {
 
 
         boolean useMeta = itemStack.hasItemMeta() == toCompare.hasItemMeta() && itemStackMeta != null, useBookMeta = itemStackBookMeta != null && toCompareBookMeta != null;
+        boolean compareFireworkDuration = Setting.FIREWORK_COMPARE_DURATION.getBoolean();
+        boolean compareFireworkEffects = Setting.FIREWORK_COMPARE_EFFECTS.getBoolean();
```

Comment:
These should also be editable per shop in EditCommand.java, the last one currently in there is on line 495. I would say to remove the main setting , but if you would like to keep that then we may want to make the rest of the shop options toggle-able in the settings as well.

Reply:
PS. I may clean up EditCommand before you get to it so the line number may not be good anymore

Reply:
It wasn't added in back then because I felt it would open a door for abuse as in shop owners scamming buyers by giving them short-duration or effect-less fireworks.
Then again, if buyers can view the shop settings, that would shift the responsibility to them.